### PR TITLE
feature(ui-ux): `portfolio` - sorting and filtering tokens

### LINKED ISSUE #2337: `portfolio` - filtering based on token type
#### What would you like to be added:
On the portfolio page, we would like to have a filtering based on the token type.
image

- [ ] All Tokens
- [ ] LP Tokens (Display all liquidity pool tokens)
- [ ] Crypto (isDAT tokens)
- [ ] dTokens (isLoanToken)

Some tokens, can appear on multiple filters. E.g, DUSD can fall under Assets/dTokens.

### LINKED ISSUE #2338: `portfolio` - sorting tokens
#### What would you like to be added:
Ability to sort tokens based on these criterias:

- [ ] Highest Value
- [ ] Lowest value

image

@chloezxyy Let's use Highest Value/Lowest Value. Let's not add USD as future iterations can be switched to BTC, Euro etc.

### PR DESCRIPTION
**Pull request by chloezxyy**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Sorting and filtering tokens based on asset type and amount on Portfolio page

#### Which issue(s) does this PR fixes?:

Fixes #2337 
Fixes #2338

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [x] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [x] Added e2e tests*
- [x] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceCard.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
+import {
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceCard.tsx:1]: can you fix format of this

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceCard.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      </ThemedTouchableOpacity>
+    )
+  }
+
+  function DropdownArrow ({
+    isExpanded,
+    setIsExpanded
+  }: { isExpanded: boolean, setIsExpanded: (isExpanded: boolean) => void }): JSX.Element {
+    return (
+      <View style={tailwind('px-4 flex flex-row items-center')}>
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceCard.tsx:171]: Make the entire component clickable

**Review by thedoublejay — CHANGES_REQUESTED** [review]

**Comment by pierregee** [comment]
image

Should we tweak the wording a little bit to be more meaningful in respect to the tab selected? CC: @thedoublejay

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceCard.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    ]
+    const onButtonGroupChange = (buttonGroupTabKey: ButtonGroupTabKey): void => {
+      if (buttonGroupOptions !== undefined) {
+        buttonGroupOptions.setActiveButtonGroup(buttonGroupTabKey)
+        buttonGroupOptions.onButtonGroupPress(buttonGroupTabKey)
+      }
+    }
+
+    const { hasFetchedToken } = useSelector((state: RootState) => (state.wallet))
+    const [isExpanded, setIsExpanded] = useState<boolean>(false)
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceCard.tsx:78]: Can change naming of this? isExpanded seems like it's for the`collapse`

File: mobile-app/cypress/integration/functional/wallet/balances/balances.spec.ts [hunk 4]
```diff
@@ -541,22 +596,87 @@ context('Wallet - Balances - USD Value', () => {
   })
 })
 
-context('Wallet - Balances - display sorted USD values', function () {
+context('Wallet - Balances - Assets filter tab', function () {
```
Comment (pierregee) [inline mobile-app/cypress/integration/functional/wallet/balances/balances.spec.ts:599]: Can add tests for sorting asc/decending too?

**Review by pierregee — CHANGES_REQUESTED** [review]

File: mobile-app/cypress/integration/functional/wallet/balances/balances.spec.ts (hunk 4 above)
Comment (pierregee) [reply mobile-app/cypress/integration/functional/wallet/balances/balances.spec.ts:599]: Already have

**Pull request closed without merging**